File: backend/app/schemas/earnings.py
"""Earnings estimate schemas."""

from datetime import date, datetime

from pydantic import BaseModel


class EarningsEstimateResponse(BaseModel):
    id: int
    earnings_date: date
    fiscal_quarter: str | None
    estimated_eps: float | None
    actual_eps: float | None
    surprise_pct: float | None
    estimated_revenue: float | None
    actual_revenue: float | None
    revenue_surprise_pct: float | None
    guidance_change: str | None
    reported: bool
    fetched_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/earnings.ts
export interface EarningsEstimateResponse {
  id: number;
  earnings_date: string;
  fiscal_quarter: string | null;
  estimated_eps: number | null;
  actual_eps: number | null;
  surprise_pct: number | null;
  estimated_revenue: number | null;
  actual_revenue: number | null;
  revenue_surprise_pct: number | null;
  guidance_change: string | null;
  reported: boolean;
  fetched_at: string;
}
